docs/build-mvm/how-to/schedule_functions.md

```diff
@@ -1 +1,883 @@
+    	pop_mailbox() -> PopMailboxResult
+    	-- Get all messages from the mailbox in FIFO order.
+    	get_mailbox() -> GetMailboxResult
+    ```
+    
+- Example
+    
+    The Spell Script:
+    
+    ```aqua
+    aqua MyFirstSpell
+    
+    export spell
+    
+    import "@fluencelabs/aqua-lib/builtin.aqua"
+    import "@fluencelabs/spell/spell_service.aqua"
+    
+    func spell():
+      Spell "self"
+    
+      mailbox_result <- Spell.pop_mailbox()
+      if mailbox_result.success && !mailbox_result.absent:
+        message = mailbox_result.message!
+        log_msg <- Debug.stringify(["Got message:", message.message, "from:", message.init_peer_id])
+        Spell.store_log(log_msg)
+    ```
+    
+    This script will add a log each time it finds the message in the mailbox.
+    
+    We can put the message ourselves via Fluence CLI. Add this into your `src/aqua/main.aqua` :
+    
+    ```aqua
+    aqua Main
+    
+    import "@fluencelabs/aqua-lib/builtin.aqua"
+    import "@fluencelabs/aqua-lib/subnet.aqua"
+    import "@fluencelabs/spell/spell_service.aqua"
+    
+    use "deals.aqua"
+    
+    export testSpell
+    
+    func testSpell():
+        deals <- Deals.get()
+        dealId = deals.dealName!.dealIdOriginal
+    
+        on HOST_PEER_ID:
+            subnet <- Subnet.resolve(dealId)
+            if subnet.success == false:
+                Console.print(["Failed to resolve subnet: ", subnet.error])
+    
+            for w <- subnet.workers:
+                on w.worker_id! via w.host_id:
+                    Spell "myFirstSpell"
+                    Spell.push_mailbox("hello from the client!")
+    ```
+    
+    And call the Fluence CLI command: `fluence run -f 'testSpell()' -i src/aqua/main.aqua` 
+    When the spell wakes up, it’ll see the message and notify about it in the log. We can check the log using the usual Fluence CLI command.
+    
+    - Output
+        
+        ```bash
+        $ fluence deal logs dealName myFirstSpell
+        Connecting to random local relay: /ip4/127.0.0.1/tcp/9992/ws/p2p/12D3KooWMNJvkaLpUKzK64CgX1x9PNdy3vLCWFgLSpB2S7ymVhEC
+        Connected
+        dealName (host_id: 12D3KooWFS4WXar3f1SWCykTUy9cVKNU8x1yDA18ZYp86mXTUyAe, worker_id: 12D3KooWBPfFGP7y1jiKgQ4DEkuS87KsfcPt5DDVoWUiwNtbzYFE, spell_id: 8e00fe09-b6d8-4c81-b925-e3fb82d2317c):
+        
+        2024-02-09 16:18:13 Got message: hello from the client! from: 12D3KooWQc1TuBEi54HNVuerVSZAMD3sDd4ZzTLan4tAgNp9SHi7
+        ```
+        
+
+### Other Spell Service API Functions
+
+If you read the full Spell Service definition, you may notice other functions not covered in the sections above. The main reason for this is that they are mostly used for utility purposes like initialization, AIR error handling, and so on, and you, as a developer, likely never need them. If you are interested, the implementation details you may find in [the Spell Service FLIP.](https://www.notion.so/Spells-Service-99e17c405f2548efb20e86866298a912?pvs=21)
```

Comment:
Are FLIPs publicly available? Which link to use?

Reply:
don't think so. may be best to cut paragraph for now.